### ISSUE TITLE
Pull in variables defined from inherited classes for top_scope_variables check

### ISSUE THREAD

**Issue by rodjek** [rodjek/puppet-lint#304]
``` puppet
class foo {
  $a = 1
}

class bar inherits foo {
  notify { $a: }
}
```

Shouldn't throw an error.

**Comment by dioni21** [comment]
Is it legit?

According to https://docs.puppetlabs.com/puppet/latest/reference/lang_classes.html#inheritance

"The base class becomes the parent scope of the derived class, so that the new class receives a copy of all of the base class’s variables and resource defaults."

Can we please reopen this issue, or should I open a new one?
